geoprocessing-tests/geocoding.spec.ts

```diff
@@ -27,62 +30,68 @@ test.describe("Geocoding Tests", () => {
-
-          // Basic validation - you can expand this based on your requirements
-          expect(responseDataLeg).toBeDefined();
-          expect(responseDatak8).toBeDefined();
-
+          var responses: number[] = [];
+          for(const index in domains){
+            
+            // Append the API Key query param to the query URL. Pull this from your environment variables and default to 'demo' test key.
+            // console.log(process.env);
+            const fullQuery = domains[index]+testCase.query;
+            //console.log(fullQuery);
+            const queryWithApiKey = new URL(fullQuery);
+            queryWithApiKey.searchParams.append("apikey", apiKey);
+
+            // DEBUG:
+            // Log the full query URL for debugging
+            // console.log(`query URL: ${queryWithApiKey.toString()}`);
+
+            // Make the geocoding API request
+            const response = await page.request.get(queryWithApiKey.toString());
+
+            // Test for 200 OK response. This doesn't guarantee the query is valid, but it's a good start.
+            expect(response.ok()).toBeTruthy();
+
+            // Because GSVCS API's don't conform exactly to any standard, the request might return a 200 OK status even if the query is invalid or returns no results.
+            // So we check the status code explicitly
+            const responseData = await response.json();
+            
+            // Basic validation - you can expand this based on your requirements
+            expect(responseData).toBeDefined();
+
+            // Add the response to a running array of all queries
+            responses.push(responseData);
+          }
+          
           // Check if testCase has attributeMatchers. If none, skip the testing and assume truthy response.
           // If testCase does have matchers, loop through them and validate each one.
-          if (
-            testCase.attributeMatchers &&
-            testCase.attributeMatchers.length > 0
-          ) {
-             assertAttributeMatchers(
-              responseDataLeg,
-              responseDatak8,
-              testCase.attributeMatchers,
-              expect
-            );
+          var success: boolean[] = [];
+          if (testCase.attributeMatchers && testCase.attributeMatchers.length > 0){   
+            for (const index in responses){
+              success.push(
+                assertAttributeMatchers(
+                  responses[0], //Set as the expectation for responses[index]
+                  responses[index],
+                  domains[index],
+                  testCase.attributeMatchers
+                )
+              )
+            }
           } else {
             console.log(
               "No attribute matchers defined for this test case, skipping validation."
             );
           }
 
+          for(const index in success){
+            expect(success[index]).toBeTruthy();
+          }
```

Comment:
Confused as to what this does. After twice evaluating `value1` and `value2`, you seemingly push the true/false results into an array and then expect those to be truthy values. 

1) `assertAttributeMatchers()` no longer asserts anything, so change function name more accurate describe what it does
2) The enclosing loop that calls `expect()` should not be called after the two if-statement conditions because it means it will execute for both cases where attribute matchers exist and the length is greater than 0 AND when that condition is false. When `false`, there are no entries in `success` because they haven't been pushed.